src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -291,5 +272,29 @@ public IChangeToken Watch(string filter)
 
             return _filesWatcher.CreateFileChangeToken(filter);
         }
+
+        private static IDirectoryContents GetEnumerableContents(string fullPath)
+        {
+            return new EnumerableDirectoryContents(() =>
```

Comment:
I considered that too and started an implementation with that approach, but ended up with the current implementation. There are probably a dozen ways it could be done, but I landed on this. I couldn't see a strong case for adding a new type such as `PhysicalDirectoryContents`.